pkg/agent/endpoints/sds/handler.go

```diff
@@ -286,9 +288,10 @@ func (h *Handler) buildResponse(versionInfo string, req *api_v2.DiscoveryRequest
 	}
 
 	// TODO: verify the type url
+	_, defaultBundleNameBool := names[h.c.SDSDefaultBundleName]
 
-	if upd.Bundle != nil && (len(names) == 0 || names[upd.Bundle.TrustDomainID()]) {
-		validationContext, err := buildValidationContext(upd.Bundle)
+	if upd.Bundle != nil && (len(names) == 0 || defaultBundleNameBool || names[upd.Bundle.TrustDomainID()]) {
```

Comment:
The `defaultBundleNameBool`/`defaultSVIDNameBool` feel a little unwieldy to me, particularly being passed as a parameter to the respective `build*` methods.

How would you feel about the following refactoring:

```
	if upd.Bundle != nil {
		switch {
		case len(names) == 0 || names[upd.Bundle.TrustDomainID()]:
			validationContext, err := buildValidationContext(upd.Bundle, "")
			if err != nil {
				return nil, err
			}
			resp.Resources = append(resp.Resources, validationContext)
		case names[h.c.SDSDefaultBundleName]:
			validationContext, err := buildValidationContext(upd.Bundle, h.c.SDSDefaultBundleName)
			if err != nil {
				return nil, err
			}
			resp.Resources = append(resp.Resources, validationContext)
		}
	}
```
Then the `build*` methods can take a single string parameter for the `nameOverride` and if empty, the name comes from the bundle/certificate.